search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java

```diff
@@ -519,6 +519,19 @@ void shouldQueryingFundingSuccessfully() throws InterruptedException, ApiGateway
             assertThat(response.getHits(), hasSize(2));
         }
 
+        @Test
+        void shouldQueryingHasFileSuccessfully() throws InterruptedException, ApiGatewayException {
+            addDocumentsToIndex("sample_publication_with_affiliations.json",
+                                "sample_publication_with_several_of_the_same_affiliation.json");
+
+            var query = queryWithTermAndAggregation(
+                "associatedArtifacts.type:\"PublishedFile\" AND associatedArtifacts.administrativeAgreement:\"false\"",
+                ApplicationConstants.RESOURCES_AGGREGATIONS);
+
+            var response = searchClient.searchWithSearchDocumentQuery(query, indexName);
+            assertThat(response.getHits(), hasSize(1));
```

Comment:
Your testing search here - that you can search for the documents with the fields you have created aggregations for. Not that the returned response from opensearch actually returns the aggregation itself. Look into the function "assertAggregation" in opensearchtest.java

Reply:
Done.  But here no need to use assertAggregation as we do not have buckets. w e have only docCount.